### ISSUE TITLE
Can't find headers while building numpy 1.16.0

### ISSUE THREAD

**Issue by V-Kh** [numpy/numpy#12839]
OK, folks, it seems that I have some issues with the most recent release. 

When I try to build something simple, for example,
### Reproducing code example:

```
python setup.py build_ext --inplace
```
I get regular output with a bunch of errors, it seems that some headers (at least) can't be found, for example, sys/endian.h

### Error message:
_configtest.c:1:10: fatal error: sys/endian.h: No such file or directory

### Numpy/Python version information:
I have a machine with Arch installed, glibc 2.28 and gcc/gfortran 8.2.1, numpy 1.16.0, Python 3.7.2

[build.log](https://github.com/numpy/numpy/files/2789587/build.log)

**Comment by V-Kh** [comment]
> Does NumPy still build despite the errors? 

It does, however, I didn't test it thoroughly.

> Are you sure you didn't observe this when building from source in 1.15.x? 

I did use a package from the repository before so I can't say there were any problems. However, I've tried to build 1.15.0 and got something similar.

I mean, I don't really care about all the errors and warnings as long as they don't break the build. However, I do plan to build numpy using icc next and I want to be sure that even basic configuration works OK.
